qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,542 @@
+import numpy as np
+from typing import Any, Tuple, Dict, Union
+
+from qcodes import (VisaInstrument, InstrumentChannel, Parameter,
+                    ParameterWithSetpoints, ChannelList)
+from qcodes.instrument.parameter import ParamRawDataType
+from qcodes.utils.validators import Enum, Numbers, Arrays, Ints
+from qcodes.utils.helpers import create_on_off_val_mapping
+
+
+class FrequencyAxis(Parameter):
+
+    def __init__(self, start: float, stop: float, npts: int, *args: Any,
+                 **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._start: float = start
+        self._stop: float = stop
+        self._npts: int = npts
+
+    def get_raw(self) -> ParamRawDataType:
+        return np.linspace(self._start, self._stop, self._npts)
+
+
+class Trace(ParameterWithSetpoints):
+
+    def __init__(self, number: int, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.instrument: Union["SpectrumAnalyzer", "PhaseNoise"]
+        self.root_instrument: "N9030B"
+
+        self.n = number
```

Comment:
```suggestion
        self.number = number
```
Try to avoid using single letter variable names.